### PR TITLE
Cleaned some code and minor fixes

### PR DESCRIPTION
**Pull request by emilwestergren**

### CODE REVIEW

**Comment by emilwestergren** [comment]
Adapted test.

**Comment by emilwestergren** [comment]
Adapted test again.

**Comment by emilwestergren** [comment]
Updated.

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/gpu/GpuImage.ooc [hunk 1]
```diff
@@ -41,17 +41,15 @@ GpuImage: abstract class extends Image {
 	_channels: Int
 	channels: Int { get { this _channels } }
 	length: Int { get { this _channels * this size width * this size height } }
-
+	_recycable := false
+	recycable ::= this _recycable
```
Comment (davidhesselbom) [inline source/draw/gpu/GpuImage.ooc]: recyc**l**able

File: source/draw/gpu/GpuYuv420Planar.ooc [hunk 2]
```diff
@@ -25,11 +25,19 @@ GpuYuv420Planar: abstract class extends GpuPlanar {
...
-		this _u free()
-		this _v free()
+	init: func (y: GpuMonochrome, u: GpuMonochrome, v: GpuMonochrome, context: GpuContext) {
+		super(y size, context)
+		this _y = y
+		this _y referenceCount increase()
+		this _u = u
+		this _u referenceCount increase()
+		this _v = v
+		this _v referenceCount increase()
```
Comment (davidhesselbom) [inline source/draw/gpu/GpuYuv420Planar.ooc]: Do `=_y` etc. instead, can skip `this _y = y`...

File: source/draw/gpu/GpuYuv420Semiplanar.ooc [hunk 3]
```diff
@@ -24,10 +24,16 @@ GpuYuv420Semiplanar: abstract class extends GpuPlanar {
...
-	init: func (size: IntSize2D, context: GpuContext) { super(size, context) }
-	dispose: func {
-		this _y free()
-		this _uv free()
+	init: func (y: GpuMonochrome, uv: GpuUv, context: GpuContext) {
+		super(y size, context)
+		this _y = y
+		this _y referenceCount increase()
+		this _uv = uv
+		this _uv referenceCount increase()
```
Comment (davidhesselbom) [inline source/draw/gpu/GpuYuv420Semiplanar.ooc]: Same here

File: source/draw/gpu/android/AndroidContext.ooc [hunk 4]
```diff
@@ -211,6 +180,6 @@ AndroidContextManager: class extends GpuContextManager {
 
 		result
 	}
-	createBgra: func ~fromGraphicBuffer (buffer: GraphicBuffer) -> OpenGLES3Bgra { this currentAndroidContext createBgra(buffer) }
-	unpackBgraToYuv420Semiplanar: func (source: GpuBgra, targetSize: IntSize2D) -> GpuYuv420Semiplanar { this currentAndroidContext unpackBgraToYuv420Semiplanar(source, targetSize) }
+	createBgra: func ~fromGraphicBuffer (buffer: GraphicBuffer) -> OpenGLES3Bgra { this currentContext createBgra(buffer) }
+	unpackBgraToYuv420Semiplanar: func (source: GpuBgra, targetSize: IntSize2D) -> GpuYuv420Semiplanar { this currentContext unpackBgraToYuv420Semiplanar(source, targetSize) }
```
Comment (davidhesselbom) [inline source/draw/gpu/android/AndroidContext.ooc]: This is a really long line

File: source/draw/gpu/android/GpuPacker.ooc [hunk 5]
```diff
@@ -60,8 +60,9 @@ GpuPacker: class {
 		this _packFence sync()
 		This flush()
 	}
+	wait: func { this _packFence clientWait(1000000000) }
```
Comment (davidhesselbom) [inline source/draw/gpu/android/GpuPacker.ooc]: Can do `100_000_000` or whatever

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/gpu/opengl/OpenGLES3Yuv420Planar.ooc [hunk 6]
```diff
@@ -18,30 +18,25 @@
...
 		result
 	}
 	create: static func ~empty (size: IntSize2D, context: GpuContext) -> This {
-		result := This new(size, context)
-		result _y = OpenGLES3Monochrome create(size, context)
-		result _u = OpenGLES3Monochrome create(IntSize2D new (size width, size height / 4), context)
-		result _v = OpenGLES3Monochrome create(IntSize2D new (size width, size height / 4), context)
+		y := context createMonochrome(size) as OpenGLES3Monochrome
+		u := context createMonochrome(IntSize2D new(size width / 4, size height / 2)) as OpenGLES3Monochrome
+		v := context createMonochrome(IntSize2D new(size width / 4, size height / 2)) as OpenGLES3Monochrome
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3Yuv420Planar.ooc]: `size width / 2`, not `4`.

**Comment by emilwestergren** [comment]
Updated.

**Pull request merged**